### PR TITLE
builtin: Implement min/max builtin function

### PR DESCRIPTION
**Pull request by corona10**
- Implement builtin min/max function
- Support the keyword-only option for `ParseTupleAndKeywords`

### CODE REVIEW

**Comment by corona10** [comment]
@ncw PTAL

**Comment by ncw** [comment]
Sorry, I got half way through reviewing this and got distracted!

File: builtin/builtin.go [hunk 1]
```diff
@@ -772,6 +772,93 @@ func builtin_len(self, v py.Object) (py.Object, error) {
...
+
+	if positional > 1 {
+		values = args
+	} else {
+		err := py.UnpackTuple(args, nil, name, 1, 1, &values)
+		if err != nil {
+			return nil, err
+		}
+	}
+	err := py.ParseTupleAndKeywords(nil, kwargs, format, kwlist, &keyFunc, &defaultValue)
```
Comment (ncw) [inline builtin/builtin.go]: I note that you haven't implemented the `default` or `key` arguments...

The default argument would be very easy to implement, just copy the `defaultValue` to `maxval` if it is not None.

The `key` argument is a little trickier...

File: builtin/builtin.go [hunk 2]
```diff
@@ -772,6 +772,93 @@ func builtin_len(self, v py.Object) (py.Object, error) {
...
+		item, err := py.Next(iter)
+		if err != nil {
+			if py.IsException(py.StopIteration, err) {
+				break
+			}
+			return nil, err
+		}
+		if maxval == nil {
+			//FIXME(@corona10): Should use keyfunc value
+			maxval = item
```
Comment (ncw) [inline builtin/builtin.go]: Duplicated line here!

File: builtin/builtin.go [hunk 3]
```diff
@@ -772,6 +772,93 @@ func builtin_len(self, v py.Object) (py.Object, error) {
...
+	for {
+		item, err := py.Next(iter)
+		if err != nil {
+			if py.IsException(py.StopIteration, err) {
+				break
+			}
+			return nil, err
+		}
+		if maxval == nil {
+			//FIXME(@corona10): Should use keyfunc value
```
Comment (ncw) [inline builtin/builtin.go]: If key is set then doing the equivalent of `maxval = key(item)` is what is required here which shouldn't be too tricky.

**Comment by corona10** [comment]
@ncw 
Looks like we have to implement keyword only arugment for `ParseTupleAndKeywords`by supporting '$'.
I will update this PR soon

**Comment by corona10** [comment]
@ncw gentle ping~ :)

**Pull request merged**